Protect runinfoPublish during resetAction

When one of the xdaq failed to init and user tries to use resetAction,
the failed xDAQ will gives a null pointer:
```
     // find all RunInfoServers controlled by this FM and acquire the information
          for (QualifiedResource qr : functionManager.containerhcalRunInfoServer.getApplications() ) {
            RISR.acquire((XdaqApplication)qr);
          }
```
A simple `isEmpty()` check should be able to protect agains this
``` 2018-03-21 15:44:31 and 584 ms : cms.hcalpro.rcms.fm.fw.user.UserEventHandler
     ERROR EventHandler: callback method resetAction failure
java.lang.reflect.InvocationTargetException
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at rcms.fm.fw.user.UserEventHandler.invokeActions(UserEventHandler.java:237)
at rcms.fm.fw.user.UserEventHandler.handleEvent(UserEventHandler.java:220)
at rcms.fm.fw.EventHandlerManager.handleEvent(EventHandlerManager.java:131)
at rcms.fm.fw.EventProcessor.processEvent(EventProcessor.java:100)
at rcms.fm.fw.FunctionManager.processEvent(FunctionManager.java:1253)
at rcms.fm.fw.FunctionManager.access$500(FunctionManager.java:51)
at rcms.fm.fw.FunctionManager$ConsumerThread.run(FunctionManager.java:1200)
Caused by: java.lang.NullPointerException
at rcms.fm.app.level1.HCALEventHandler$RunInfoServerReader.acquire(HCALEventHandler.java:1831)
at rcms.fm.app.level1.HCALEventHandler.publishRunInfoSummaryfromXDAQ(HCALEventHandler.java:1102)
at rcms.fm.app.level1.HCALlevelTwoEventHandler.resetAction(HCALlevelTwoEventHandler.java:317)```

--- a/src/rcms/fm/app/level1/HCALEventHandler.java
+++ b/src/rcms/fm/app/level1/HCALEventHandler.java
@@ -636,6 +636,9 @@ public class HCALEventHandler extends UserEventHandler {
       e.printStackTrace( new PrintWriter(sw) );
       System.out.println(sw.toString());
       //String errMessage = "[HCAL " + functionManager.FMname + "] " + this.getClass().toString() + " failed to initialize resources. Printing stacktrace: "+ sw.toString();
+      
+      //Reset runinfoserver pointer if initxdaq failed
+      functionManager.containerhcalRunInfoServer  = null;
       throw new UserActionException(e.getMessage());
     }
 

